src/skywallet/transaction.go

```diff
@@ -174,3 +193,147 @@ func (s *SkycoinTransactionSigner) addSignatures(msg *wire.Message) error {
 	}
 	return nil
 }
+
+// BitcoinTransactionSigner represents signing Skycoin transaction process
+// @used_in TransactionSign
+type BitcoinTransactionSigner struct {
+	Device     *Device
+	Inputs     []*messages.BitcoinTransactionInput
+	Outputs    []*messages.BitcoinTransactionOutput
+	Version    int
+	LockTime   int
+	signatures []string
+	state      int
+}
+
+// SetDevice assigns device, which will be signing
+func (s *BitcoinTransactionSigner) SetDevice(device *Device) {
+	s.Device = device
+}
```

Comment:
why is this even needed 
```
	signer := skyWallet.BitcoinTransactionSigner{
		Inputs:   transactionInputs,
		Outputs:  transactionOutputs,
		Version:  1,
		LockTime: 0,
	}

	signatures, err := device.GeneralTransactionSign(&signer)
	if err != nil {
		return err
	}
```

u can just do 

```
	signer := skyWallet.BitcoinTransactionSigner{
		Device:   device,
		Inputs:   transactionInputs,
		Outputs:  transactionOutputs,
		Version:  1,
		LockTime: 0,
	}

	signatures, err := device.GeneralTransactionSign(&signer)
	if err != nil {
		return err
	}
```

Reply:
The same as for TransactionSigner